worlds/sa2b/GateBosses.py

```diff
@@ -0,0 +1,69 @@
+import typing
+
+speed_characters_1 = "Sonic vs Shadow 1"
+speed_characters_2 = "Sonic vs Shadow 2"
+mech_characters_1 = "Tails vs Eggman 1"
+mech_characters_2 = "Tails vs Eggman 2"
+hunt_characters_1 = "Knuckles vs Rouge 1"
+big_foot = "F-6t BIG FOOT"
+hot_shot = "B-3x HOT SHOT"
+flying_dog = "R-1/A FLYING DOG"
+egg_golem_sonic = "Egg Golem (Sonic)"
+egg_golem_eggman = "Egg Golem (Eggman)"
+king_boom_boo = "King Boom Boo"
+
+gate_bosses_no_requirements_table = {
+    speed_characters_1: 0,
+    speed_characters_2: 1,
+    mech_characters_1: 2,
+    mech_characters_2: 3,
+    hunt_characters_1: 4,
+    big_foot: 5,
+    hot_shot: 6,
+    flying_dog: 7,
+    egg_golem_sonic: 8,
+    egg_golem_eggman: 9,
+}
+
+gate_bosses_with_requirements_table = {
+    king_boom_boo: 10,
+}
+
+all_gate_bosses_table = {
+    **gate_bosses_no_requirements_table,
+    **gate_bosses_with_requirements_table,
+}
+
+
+def get_boss_name(boss: int):
```

Comment:
could pre-compute a lookup dict boss_names = {value: key for key, value in original.items()}
Since it's only used in the spoiler output it's probably ok as-is, just a minor recommendation.